version/helper.go

```diff
@@ -64,7 +65,19 @@ func GetAgentPackageVersionFilePath() (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("detecting current executable path: %w", err)
 	}
-	return filepath.Join(filepath.Dir(execPath), PackageVersionFileName), nil
+
+	dirPath := filepath.Dir(execPath)
+
+	if runtime.GOOS == "darwin" {
+		// On Mac the path is different because of package signing issues
+		// we have to go outside the elastic-agent.app directory
+		appDirIndex := strings.Index(dirPath, "/elastic-agent.app/")
+		if appDirIndex != -1 {
+			dirPath = dirPath[:appDirIndex]
+		}
+	}
+
+	return filepath.Join(dirPath, PackageVersionFileName), nil
```

Comment:
Take a look at https://github.com/elastic/elastic-agent/blob/454e6253a0e3457828b769e7c2c7c0e9365d870b/internal/pkg/agent/application/paths/common.go#L211-L221 and also https://github.com/elastic/elastic-agent/blob/454e6253a0e3457828b769e7c2c7c0e9365d870b/internal/pkg/agent/application/paths/common.go#L229-L241 and how they two are used together in https://github.com/elastic/elastic-agent/blob/454e6253a0e3457828b769e7c2c7c0e9365d870b/internal/pkg/agent/application/paths/common.go#L206-L208.

If I'm reading the logic of this function here, it looks pretty much the same as the logic in the functions mentioned above.  So maybe you could reuse them somehow, either by exporting `retrieveExecutableDir()` from the `paths` package or by moving `GetAgentPackageVersionFilePath()` into the `paths` package?  This would also let you get rid of the `getCurrentExecutablePath` function as I believe it's only being called from the `GetAgentPackageVersionFilePath` function.

Reply:
In this patch I am not making any assumption as to what the directory structure is... I just cut off the part of the path that goes in `elastic-agent.app`
the code we have in paths requires that there are exactly 2 or 5 levels (for Mac) between the executable and the execDir.

The initial idea of this was to have the package version file side by side with the executable wherever it is... we are just moving it for mac because of package signing issues and I would prefer to keep changes to a minimum for this hotfix. We can open a follow-up issue if you feel that we should remove the duplication (we still have to verify that everything still works also for packages that we just extract somewhere on disk (it must not depend on installing agent)

Reply:
Sounds fair. Would you mind opening that follow up issue and linking it here so we don't lose track of this work? Thanks.

Reply:
Created #3337